gridstatus/ercot.py

```diff
@@ -3658,3 +3667,94 @@ def _handle_indicative_lmp_by_settlement_point(self, df: pd.DataFrame):
                 "LMP",
             ]
         ]
+
+    @support_date_range(frequency="DAY_START")
+    def get_dam_total_energy_purchased(
+        self,
+        date: str | pd.Timestamp,
+        end: str | pd.Timestamp | None = None,
+        verbose: bool = False,
+    ) -> pd.DataFrame:
+        """Get DAM Total Energy Purchased
+
+        Arguments:
+            date (str, datetime): date to get data for
+            end (str, datetime): end time to get data for
+            verbose (bool, optional): print verbose output. Defaults to False.
+
+        Returns:
+            pandas.DataFrame: A DataFrame with DAM total energy purchased data
+        """
+        if date == "latest":
+            return self.get_dam_total_energy_purchased(
+                date="today",
+                verbose=verbose,
+            )
+
+        # DAM data so subtract one from the date
```

Comment:
For example, to get data for May 20, we want to retrieve the file from May 19. The function returns data on the provided `date`.